specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/stable/2017-11-15/timeseriesinsights.json

```diff
@@ -1365,9 +1372,36 @@
             "name": "StorageLimitExceededBehavior",
             "modelAsString": false
           }
+        },
+        "partitionKeyProperties": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EnvironmentInfoProperty"
+          },
+          "description": "The list of event properties which will be used to partition data in the environment."
         }
       }
     },
+    "EnvironmentInfoProperty": {
```

Comment:
I think I would make this PartitionKeyProperty instead of trying to make it generic.